### PR TITLE
NavGraph - SV Audit 

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
PR to add audit feedback changes for the `NavGraph`

## Links and Data
Sample Epic: [#4824](https://devtopia.esri.com/runtime/kotlin/issues/4824)

### CODE REVIEW

**Comment by shubham7109** [comment]
@eri9000 I have added a [commit](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/247/commits/963fb78b654408afcbb65373aa4b56b43e049922) to implement a fix for the `popBackStack`. If the back button is spammed there could be a scenario where `popBackStack` is called multiple times, and with no items in the back stack, compose will display a blank screen. 

In order to avoid this, I added an extension function to check if navigation to a destination is possible or `navigateToHome` if no destination found.

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/navigation/NavGraph.kt [hunk 1]
```diff
@@ -65,7 +56,7 @@ fun NavGraph() {
         }
 
         composable(Routes.ABOUT_SCREEN) {
-            AboutScreen()
+            AboutScreen(popBackStack = { navController.pop() })
```
Comment (eri9000) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/navigation/NavGraph.kt:59]: parameter name sounds more descriptive 
```suggestion
            AboutScreen(onBackPressed = { navController.pop() })
```

**Review by eri9000 — COMMENTED** [review]
Overall look good,
some suggestion s

**Comment by shubham7109** [comment]
Thanks @eri9000, ready for you re-review!

**Review by eri9000 — APPROVED** [review]
LGTM 👍🏼

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/about/AboutScreen.kt [hunk 2]
```diff
@@ -507,5 +507,5 @@ private fun ApiDetailsSection() {
 @Preview(uiMode = Configuration.UI_MODE_NIGHT_YES, showBackground = true)
 @Composable
 fun PreviewAboutScreen() {
-    SampleAppTheme { AboutScreen() }
+    SampleAppTheme { AboutScreen {} }
```
Comment (TADraeseke) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/about/AboutScreen.kt:510]: not sure this needs curly brackets

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -22,4 +22,5 @@
     <string name="clearButton">Clear button</string>
     <string name="cancel">Cancel</string>
     <string name="sdk_sample_icon_description">SDK sample icon</string>
+    <string name="home_screen_title">Sample Categories</string>
 </resources>
```
Comment (TADraeseke) [inline app/src/main/res/values/strings.xml:26]: new line at end of file

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -22,4 +22,5 @@
     <string name="clearButton">Clear button</string>
     <string name="cancel">Cancel</string>
     <string name="sdk_sample_icon_description">SDK sample icon</string>
+    <string name="home_screen_title">Sample Categories</string>
```
Comment (TADraeseke) [inline app/src/main/res/values/strings.xml:25]:
```suggestion
    <string name="home_screen_title">Sample categories</string>
```

**Review by TADraeseke — APPROVED** [review]
LGTM! Just a few small bits

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/about/AboutScreen.kt (hunk 2 above)
Comment (shubham7109) [reply app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/ui/screens/about/AboutScreen.kt:510]: We don't need to support the `onBackPressed` in a preview, so I've left it as an empty lambda

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (shubham7109) [reply app/src/main/res/values/strings.xml:25]: I feel like this change goes against the other top app bar titles which are capitalized (eg: `Augmented Reality`,`Utility Networks`

Keeping it as is, unless you would strongly recommend :)

**Pull request merged**